src/plugins/application_config/server/routes/index.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  IRouter,
+  IScopedClusterClient,
+  Logger,
+  OpenSearchDashboardsResponseFactory,
+} from '../../../../core/server';
+import { ConfigurationClient } from '../types';
+
+export function defineRoutes(
+  router: IRouter,
+  getConfigurationClient: (inputOpenSearchClient: IScopedClusterClient) => ConfigurationClient,
+  logger: Logger
+) {
+  router.get(
+    {
+      path: '/api/config/exists',
+      validate: false,
+    },
+    async (context, request, response) => {
+      const client = getConfigurationClient(context.core.opensearch.client);
+
+      return await handleExistsConfig(client, response, logger);
+    }
+  );
+}
+
+export async function handleExistsConfig(
+  client: ConfigurationClient,
+  response: OpenSearchDashboardsResponseFactory,
+  logger: Logger
+) {
+  try {
+    const result = await client.existsConfig();
+    return response.ok({
+      body: {
+        exists: result,
+      },
+    });
+  } catch (e) {
+    logger.error(e);
+    return errorResponse(response, e);
+  }
+}
+
+export function errorResponse(response: OpenSearchDashboardsResponseFactory, error: any) {
```

Comment:
This looks like a private function, do we need to export it? Also wonder if we can have response object which defines the structure similar how other Osd errors, possibly by extending OsdError.

Reply:
Originally it was for ease of unit testing to boost the test coverage to pass in Github workflow. I could actually indirectly test this function as part of other functions. I will update to not export it.

Looks like the `response` has a built in function `customError` for creating error response. I will switch to use it.